examples/go/filewatcher/cmd/root.go

```diff
@@ -0,0 +1,27 @@
+package cmd
+
+import (
+	"errors"
+	"flag"
+)
+
+type Config struct {
+	Dir     string
+	APIURL  string
+	Verbose bool
+}
+
+func ParseFlags() (*Config, error) {
+	cfg := &Config{}
+
+	flag.StringVar(&cfg.Dir, "dir", "", "Directory to watch")
+	flag.StringVar(&cfg.APIURL, "api-url", "", "API endpoint")
```

Comment:
HTTP API is not a bad choice tbh but given that we expect the watcher to run on the same system as the future components (atleast for this example), how about a unix domain socket path?

Reply:
I think this is deprecated now given the structure below